ElectronClient/app/gui/ResourceScreen.tsx

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable enforce-react-hooks/enforce-react-hooks */
+import * as React from 'react';
+
+const { connect } = require('react-redux');
+const { _ } = require('lib/locale.js');
+const { themeStyle } = require('../theme.js');
+const { bridge } = require('electron').remote.require('./bridge');
+const { Header } = require('./Header.min.js');
+const prettyBytes = require('pretty-bytes');
+const Resource = require('lib/models/Resource.js');
+
+interface Props {
+	style: any;
+	theme: any;
+}
+
+interface Resource {
+	title: string
+	id: string
+	size: number
+	file_extension: string
+	daoResource: any
+}
+
+interface State {
+	resources: Resource[] | undefined
+	sorting: ActiveSorting
+	isLoading: boolean
+}
+
+interface ResourceTable {
+	resources: Resource[]
+	sorting: ActiveSorting
+	onResourceClick: (resource: Resource) => any
+	onResourceDelete: (resource: Resource) => any
+	onToggleSorting: (order: SortingOrder) => any
+}
+
+type SortingOrder = 'size' | 'name'
+type SortingType = 'asc' | 'desc'
+
+interface ActiveSorting {
+	order: SortingOrder
+	type: SortingType
+}
+
+const ResourceTable: React.FC<ResourceTable> = (props: ResourceTable) => {
+	const sortOrderEngagedMarker = (s: SortingOrder) => {
+		return (
+			<a href="#" onClick={ () => props.onToggleSorting(s) }>{
+				(props.sorting.order === s && props.sorting.type === 'desc') ? '▾' : '▴' }</a>
+		);
+	};
+	return <table style={{ width: '90%' }}>
+		<thead>
+			<tr>
+				<th>{_('Title')} { sortOrderEngagedMarker('name') }</th>
+				<th>{_('Size')} { sortOrderEngagedMarker('size') }</th>
+				<th>{_('ID')}</th>
+				<th>{_('Action')}</th>
+			</tr>
+		</thead>
+		<tbody>
+			{props.resources.map((resource: Resource, index: number) =>
+				<tr key={index}>
+					<td>
+						<a href="#" onClick={() => props.onResourceClick(resource)}>{resource.title}</a>
+					</td>
+					<td>{prettyBytes(resource.size)}</td>
+					<td>{resource.id}</td>
+					<td>
+						<button onClick={ () => props.onResourceDelete(resource) }>{_('Delete')}</button>
+					</td>
+				</tr>
+			)}
+		</tbody>
+	</table>;
+};
+
+const getSortingOrderColumn = (s: SortingOrder): string => {
+	switch (s) {
+	case 'name': return 'title';
+	case 'size': return 'size';
+	}
+};
+
+const getNextSortingOrderType = (s: SortingType): SortingType => {
+	if (s === 'asc') {
+		return 'desc';
+	} else {
+		return 'asc';
+	}
+};
+
+const MAX_RESOURCES = 10000;
+
+class ResourceScreenComponent extends React.Component<Props, State> {
+	constructor(props: Props) {
+		super(props);
+		this.state = {
+			resources: undefined,
+			sorting: {
+				type: 'asc',
+				order: 'name',
+			},
+			isLoading: false,
+		};
+	}
+
+	async reloadResources(sorting: ActiveSorting) {
+		this.setState({ isLoading: true });
+		const resources = await Resource.modelSelectAll(
+			`SELECT title, id, size, file_extension FROM resources ORDER BY ${
+				getSortingOrderColumn(sorting.order)} ${sorting.type} LIMIT ${MAX_RESOURCES}`);
```

Comment:
Please move the SQL to models/Resource

Reply:
That should work actually, without modifying models/Resource:

```
const resources = await Resource.all({
	order: [{
		by: getSortingOrderColumn(sorting.order),
		dir: sorting.type,
	}],
	limit: MAX_RESOURCES,
	fields: ['title', 'id', 'size', 'file_extension'],
})
```

Not tested but see BaseModel.all and BaseModel.applySqlOptions or more information.